### PR TITLE
Update Configuration Component

### PR DESCRIPTION
**Pull request by MLopezJ**
More info: #256

### CODE REVIEW

File: src/components/Asset/Settings/textAsButton.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}: {
+	role: string
+	tabIndex: number
+	onClick: () => any
+	onKeyPress: () => any
+	children: React.ReactNode
+	className?: string
+}) => {
+	return (
+		<u
```
Comment (coderbyheart) [inline src/components/Asset/Settings/textAsButton.tsx]: Use `button.link`

File: src/components/Asset/Settings/Settings.module.css [hunk 2]
```diff
@@ -65,8 +65,28 @@
...
 }
 
-.FooterWithFullWidthButton {
+.FullWidth {
   grid-column: auto / span 2;
   display: flex;
   flex-direction: column;
 }
+
+.collap {
```
Comment (coderbyheart) [inline src/components/Asset/Settings/Settings.module.css]:
```suggestion
.collapse {
```

File: src/components/Asset/Settings/Settings.module.css [hunk 3]
```diff
@@ -65,8 +65,28 @@
...
+  margin-bottom: 1rem;
+  padding: 1rem;
+  background-color: #e9ecef;
+  background-color: #f5f5f5;
+  border-radius: 5px;
+}
+
+#rowOne {
```
Comment (coderbyheart) [inline src/components/Asset/Settings/Settings.module.css]: Use selectors, not IDs

File: src/components/Asset/Settings/Presets.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	>
+	currentDesiredConfig: Static<typeof AssetConfig>
+}) => {
+	const presetConfig = (id: string) => {
+		const config =
+			presetConfigs[`${id}`] !== undefined
+				? presetConfigs[`${id}`].config
+				: currentDesiredConfig
+		setDesiredConfig(config)
+	}
```
Comment (coderbyheart) [inline src/components/Asset/Settings/Presets.tsx]: No need to check here, because we have a valid config in the loop in line 40

File: src/components/Asset/Settings/Presets.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		const currentDesiredConfig = presetConfigs.walking.config
+
+		const presets = isolateComponent(
+			<Presets
+				setDesiredConfig={setNewDesiredConfig}
+				currentDesiredConfig={currentDesiredConfig}
+			/>,
+		)
+
+		const parcel = presets.findOne('#walking-info')
```
Comment (coderbyheart) [inline src/components/Asset/Settings/Presets.spec.tsx]: Finde elements for tests using test attributes: https://playwright.dev/docs/selectors#define-explicit-contract

File: src/components/Asset/Settings/Presets.module.css [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  border-top: 1px solid rgba(0, 0, 0, 0.125);
+}
+
+.FullWidth {
+  grid-column: auto / span 2;
+  display: flex;
+  flex-direction: column;
+}
+
+.collap {
```
Comment (coderbyheart) [inline src/components/Asset/Settings/Presets.module.css]:
```suggestion
.collapse {
```

File: src/components/Asset/Settings/NumberConfigSetting.tsx [hunk 7]
```diff
@@ -65,7 +74,9 @@ export const NumberConfigSetting = ({
 					)}
 				/>
 				<input
-					className="form-control"
+					className={`form-control ${
```
Comment (coderbyheart) [inline src/components/Asset/Settings/NumberConfigSetting.tsx]: Use `classNames`

File: src/components/Asset/Settings/NumberConfigSetting.tsx [hunk 8]
```diff
@@ -26,10 +28,17 @@ export const NumberConfigSetting = ({
...
+	errorMessage?: string
+	reference?: React.LegacyRef<HTMLInputElement>
 }) => {
 	const [input, updateInput] = useState(`${desired ?? reported}`)
 	const minValue = minimum ?? 0
 	const maxValue = maximum ?? Number.MAX_SAFE_INTEGER
+
+	useEffect(() => {
+		updateInput(`${desired ?? reported}`)
+	}, [desired, reported, updateInput])
```
Comment (coderbyheart) [inline src/components/Asset/Settings/NumberConfigSetting.tsx]: Why?

File: src/asset/asset.ts [hunk 9]
```diff
@@ -47,7 +47,7 @@ export const AssetConfig = Type.Object(
 		}),
 		accath: Type.Number({
 			description:
-				'Accelerometer activity threshold in m/s²: Minimal absolute value for an accelerometer reading to be considered movement.',
+				'Accelerometer Activity Threshold in m/s²: Minimal absolute value for an accelerometer reading to be considered movement.',
```
Comment (coderbyheart) [inline src/asset/asset.ts:50]: Make consistent

**Review by coderbyheart — COMMENTED** [review]
Adding a fews things to go over together.

File: src/asset/config.ts [hunk 10]
```diff
@@ -12,3 +12,43 @@ export const defaultConfig: Static<typeof AssetConfig> = {
...
+			accath: 10, // Accelerometer activity threshold
+			accith: 5, // Accelerometer inactivity threshold
+			accito: 1200, // Accelerometer inactivity timeout
+			mvt: 21600, // Movement Timeout
+			actwt: 10,
+			gnsst: 10,
+			nod: [],
+		},
+		label: 'Parcel',
+		description: 'Used for tracking parcels.',
```
Comment (coderbyheart) [inline src/asset/config.ts]: Add more extensive description (explain what values are specifically set and why).

File: src/asset/config.ts [hunk 11]
```diff
@@ -12,3 +12,43 @@ export const defaultConfig: Static<typeof AssetConfig> = {
...
+export const presetConfigs: Record<
+	string,
+	{
+		config: Static<typeof AssetConfig>
+		label: string
+		description: string
+	}
+> = {
+	parcel: {
+		config: {
```
Comment (coderbyheart) [inline src/asset/config.ts:25]: Only make it set values that are needed for this use case

File: src/components/Asset/Settings/Presets.module.css [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  border-top: 1px solid rgba(0, 0, 0, 0.125);
+}
+
+.FullWidth {
+  grid-column: auto / span 2;
+  display: flex;
+  flex-direction: column;
+}
+
+.collapse {
```
Comment (coderbyheart) [inline src/components/Asset/Settings/Presets.module.css]: Make re-usable (it is used twice)

File: src/components/Asset/Settings/NumberConfigSetting.tsx [hunk 13]
```diff
@@ -82,6 +90,7 @@ export const NumberConfigSetting = ({
 					}}
 				/>
 				<span className="input-group-text">{unit ?? 's'}</span>
+				<div className="invalid-feedback">{errorMessage}</div>
```
Comment (coderbyheart) [inline src/components/Asset/Settings/NumberConfigSetting.tsx]: Do not render if no error message

File: src/components/Asset/Personalization.tsx [hunk 14]
```diff
@@ -26,6 +26,7 @@ export const Personalization = ({ asset }: { asset: Asset }) => {
 					style={{ width: '100%' }}
 				>
 					<div className="flex-grow-1">
+						{/** TODO: chango to 'classNames' for consisten */}
```
Comment (MLopezJ) [inline src/components/Asset/Personalization.tsx]: I notice we made the same behavior we are requesting to change (https://github.com/NordicSemiconductor/asset-tracker-cloud-app-aws-js/pull/260#discussion_r977623543) here as well, but this component is not related to this PR at all. 

Should we update this here as well (for for consistency reasons) or should we just leave it as it is?

File: src/components/Asset/Settings/textAsButton.tsx (hunk 1 above)
Comment (MLopezJ) [reply src/components/Asset/Settings/textAsButton.tsx]: Can you elaborate more? 

I think I'm not getting the idea.

Comment (MLopezJ) [reply src/components/Asset/Settings/textAsButton.tsx]: I mean, I'm not sure if this is requesting to use <a> tag with href prop or to update the TextAsButton component or something else.

**Review by MLopezJ — COMMENTED** [review]
.

File: src/components/Asset/Personalization.tsx (hunk 14 above)
Comment (coderbyheart) [reply src/components/Asset/Personalization.tsx:29]: Not related.

Comment (coderbyheart) [reply src/components/Asset/Personalization.tsx]: Unrelated.

**Comment by MLopezJ** [comment]
Doing `smoke testing` before `squash and merge`

**Comment by MLopezJ** [comment]
Found a bug during `smoke testing`. Link on `Configuration Explainer` is not working. 

Working on resolve it.
